docs/src/docs/asciidoc/gradle-plugin-quickstart.adoc

```diff
@@ -0,0 +1,467 @@
+
+== Add JUnit Testing
+
+The Gradle plugin for GraalVM Native Image can run
+https://junit.org/junit5/docs/current/user-guide/[JUnit Platform] tests on your native executable. This means that the tests will be compiled and run as native code.
+
+. Create the following test in the
+`fortune/src/test/java/demo/FortuneTest.java` file:
++
+.fortune/src/test/java/demo/FortuneTest.java
+[source,java]
+----
+package demo;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class FortuneTest {
+    @Test
+    @DisplayName("Returns a fortune")
+    void testItWorks() throws JsonProcessingException {
+        Fortune fortune = new Fortune();
+        assertTrue(fortune.randomFortune().length()>0);
+    }
+}
+----
+
+. Run JUnit tests:
+[source,shell]
+----
+./gradlew nativeTest
+----
+
+The plugin runs tests on the JVM prior to running tests from the native executable. To disable testing support (which comes by default), add the following configuration to the _build.gradle_ file:
+
+[source,groovy,subs="verbatim,attributes", role="multi-language-sample"]
+----
+graalvmNative {
+    testSupport = false
+}
+----
+
+[source,kotlin,subs="verbatim,attributes", role="multi-language-sample"]
+----
+graalvmNative {
+    testSupport.set(false)
+}
+----
+
+== Run Tests with the Agent
+
+If you need to test collecting metadata with the agent, add the
+`-Pagent` option to the `test` and `nativeTest` task invocations:
+
+. Run the tests on the JVM with the agent:
+[source,shell]
+----
+./gradlew -Pagent test
+----
+
+It runs your application on the JVM with the agent, collects the metadata and uses it for testing on `native-image`. The generated configuration files (containing the metadata) can be found in the
+_$\{buildDir}/native/agent-output/$\{taskName}_ directory. 
+In this case, the plugin also substitutes `{output_dir}` in the agent options to point to this directory.
+. Build a native executable using the metadata collected by the agent:
+
+[source,shell]
+----
+./gradlew -Pagent nativeTest
+----
+
+=== Summary
+
+The Gradle plugin for GraalVM Native Image adds support for building and testing native executables using the https://gradle.org[Gradle]. The plugin has many features, described in the 
+https://graalvm.github.io/native-build-tools/latest/gradle-plugin.html[plugin
+reference documentation].
+
+Note that if your application does not call any classes dynamically at run time, the execution with the agent is needless. Your workflow, in that case, is just:
+
+[source,shell]
+----
+./gradlew nativeRun
+----
+
+Lastly, if you use GraalVM Enterprise as your `JAVA_HOME` environment, the plugin builds a native executable with enterprise features enabled.
```

Comment:
That sounds a bit weird. Sure, you get EE optimizations if your use EE, but EE features such as SBOM need to be explicitly enabled by the user.

Reply:
It was just an argument for using EE, can be removed.